src/engine/effects/engineeffectsdelay.cpp

```diff
@@ -0,0 +1,105 @@
+#include "engine/effects/engineeffectsdelay.h"
+
+#include "util/sample.h"
+
+namespace {
+// See enginedelay.cpp
+const int kiMaxDelay = static_cast<int>(0.508 *
+        mixxx::audio::SampleRate::kValueMax * mixxx::kEngineChannelCount);
+} // anonymous namespace
+
+EngineEffectsDelay::EngineEffectsDelay()
+        : m_delaySamples(0),
+          m_oldDelaySamples(0),
+          m_delayPos(0) {
+    m_pDelayBuffer = SampleUtil::alloc(kiMaxDelay);
+}
+
+void EngineEffectsDelay::process(const CSAMPLE* M_RESTRICT pIn,
+        CSAMPLE* M_RESTRICT pOutput,
+        const int iBufferSize) {
+    if (m_oldDelaySamples == 0 && m_delaySamples == 0) {
+        SampleUtil::copy(pOutput, pIn, iBufferSize);
+        return;
+    }
+
+    if (m_oldDelaySamples == m_delaySamples) {
+        int delaySourcePos = (m_delayPos + kiMaxDelay - m_delaySamples) % kiMaxDelay;
+
+        VERIFY_OR_DEBUG_ASSERT(delaySourcePos >= 0) {
+            SampleUtil::copy(pOutput, pIn, iBufferSize);
+            return;
+        }
+        VERIFY_OR_DEBUG_ASSERT(delaySourcePos <= static_cast<int>(kiMaxDelay)) {
+            SampleUtil::copy(pOutput, pIn, iBufferSize);
+            return;
+        }
+
+        for (int i = 0; i < iBufferSize; ++i) {
+            // Put samples into delay buffer.
+            m_pDelayBuffer[m_delayPos] = pIn[i];
+            m_delayPos = (m_delayPos + 1) % kiMaxDelay;
+
+            // Take a delayed sample from the delay buffer
+            // and copy it to the destination buffer.
+            pOutput[i] = m_pDelayBuffer[delaySourcePos];
+            delaySourcePos = (delaySourcePos + 1) % kiMaxDelay;
+        }
+
+    } else {
+        VERIFY_OR_DEBUG_ASSERT(m_delaySamples >= 0) {
+            SampleUtil::copy(pOutput, pIn, iBufferSize);
+            return;
+        }
+
+        int delaySourcePos = (m_delayPos + kiMaxDelay - m_delaySamples +
+                                     iBufferSize / 2) %
+                kiMaxDelay;
```

Comment:
it looks like clang-format needs your help here:
```suggestion
        int delaySourcePos = 
                (m_delayPos + kiMaxDelay - m_delaySamples + iBufferSize / 2) %
                kiMaxDelay;
```
Does this work?

Reply:
Yes, it works.